uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/shared/components/leave-request-actions.component.js

```diff
@@ -116,13 +120,31 @@ define([
     };
 
     vm.allowedActions = [];
+    vm.loading = { component: true };
 
     vm.action = action;
     vm.openLeavePopup = openLeavePopup;
 
     (function init () {
-      indexSupportData();
-      setAllowedActions();
+      $q.resolve()
+        .then(indexSupportData)
+        .then(loadCurrentlyLoggedInContactId)
+        .then(function () {
```

Comment:
Lets try to keep the `init` function simple. So please move this whole new function block to a separate function with a meaningful name, for example `setRoleAsAdminForSelfApprovers()`

Reply:
And then `setRoleAsAdminForSelfApprovers ()` and `setAllowedActions()`  can be called in parallel.